### PR TITLE
Adding the qa generator as is

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by nagkumar91** [comment]
> Do we need to add `__init__.py` to `src/promptflow-evals/promptflow/evals/synthetic`?

Eventually, when I add the simulator, I'll add the init

**Review by ninghu — DISMISSED** [review]

File: src/promptflow-evals/tests/unittests/generated_qa_chat_conv.jsonl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{"messages":[{"role":"user","content":"What is Compute Instance?"},{"role":"assistant","content":"Compute instance is ..."}]}
```
Comment (ninghu) [inline src/promptflow-evals/tests/unittests/generated_qa_chat_conv.jsonl:1]: >> Put all test data/configs under src/promptflow/tests/test_configs.

Just read this from dev_setup.md.
https://github.com/microsoft/promptflow/blob/main/docs/dev/dev_setup.md

**Review by ninghu — DISMISSED** [review]

File: src/promptflow-evals/promptflow/evals/synthetic/qa.py [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    """Class for generating Question-Answer data from text."""
+
+    _PARSING_ERR_UNEQUAL_QA = "Parsing error: Unequal question answer count"
+    _PARSING_ERR_UNEQUAL_Q_AFTER_MOD = "Parsing error: Unequal question count after modification"
+    _PARSING_ERR_FIRST_LINE = "Parsing error: First line must be a question"
+
+    def __init__(self, model_config: Dict):
+        """Initialize QADataGenerator using Azure OpenAI details."""
+
+        api_key = "OPENAI_API_KEY"
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/synthetic/qa.py:189]: Should we move this to constants.py ?

File: src/promptflow-evals/promptflow/evals/synthetic/qa.py [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+                        # QnA columns:
+                        data_dict[question_key].append(question)
+
+                    data_dict[answer_key].append(answer)  # Consider generated answer as the ground truth
+        else:
+            for qs_and_as in results:
+                chat_history = []
+                for question, answer in qs_and_as:
+                    if qa_type == QAType.CONVERSATION:
+                        print(f"Chat data dict: {data_dict['messages']}\n\n")
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/synthetic/qa.py:367]: logger ?

File: src/promptflow-evals/promptflow/evals/synthetic/qa.py [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+                    else:
+                        messages = []
+                        messages.append({"role": "user", "content": question})
+                        messages.append({"role": "assistant", "content": answer})
+                        data_dict["messages"].append(list(messages))
+        # export to jsonl file
+        try:
+            import pandas as pd
+        except ImportError as ie:
+            print("In order to write qa data to file, please install pandas")
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/synthetic/qa.py:380]: logger for all print statements ?

**Review by singankit — APPROVED** [review]

**Review by luigiw — APPROVED** [review]

**Pull request merged**